src/pages/tasks/TaskAssigneeSelectorModal.js

```diff
@@ -0,0 +1,207 @@
+            searchValue.trim(),
+            [],
+            CONST.EXPENSIFY_EMAILS,
+            false,
+        );
+
+        setHeaderMessage(OptionsListUtils.getHeaderMessage(recentReports?.length + personalDetails?.length !== 0, Boolean(userToInvite), searchValue));
+
+        setFilteredUserToInvite(userToInvite);
+        setFilteredRecentReports(recentReports);
+        setFilteredPersonalDetails(personalDetails);
+    }, [props, searchValue]);
+
+    useEffect(() => {
+        Timing.start(CONST.TIMING.SEARCH_RENDER);
+        Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+
+        updateOptions();
+
+        return () => {
+            Timing.end(CONST.TIMING.SEARCH_RENDER);
+            Performance.markEnd(CONST.TIMING.SEARCH_RENDER);
+        };
+    }, [updateOptions]);
+
+    const onChangeText = (newSearchTerm = '') => {
+        setSearchValue(newSearchTerm);
+        updateOptions();
+    };
+
+    const sections = useMemo(() => {
+        const sectionsList = [];
+        let indexOffset = 0;
+
+        sectionsList.push({
+            title: props.translate('common.recents'),
+            data: filteredRecentReports,
+            shouldShow: filteredRecentReports?.length > 0,
+            indexOffset,
+        });
+        indexOffset += filteredRecentReports?.length;
+
+        sectionsList.push({
+            title: props.translate('common.contacts'),
+            data: filteredPersonalDetails,
+            shouldShow: filteredPersonalDetails?.length > 0,
+            indexOffset,
+        });
+        indexOffset += filteredRecentReports?.length;
+
+        if (filteredUserToInvite) {
+            sectionsList.push({
+                data: [filteredUserToInvite],
+                shouldShow: filteredUserToInvite?.length > 0,
+                indexOffset,
+            });
+        }
+
+        return sectionsList;
+    }, [filteredPersonalDetails, filteredRecentReports, filteredUserToInvite, props]);
+
+    const selectReport = (option) => {
+        if (!option) {
+            return;
+        }
+
+        if (option.alternateText) {
+            // Clear out the state value, set the assignee and navigate back to the NewTaskPage
+            setSearchValue('');
+            TaskUtils.setAssigneeValue(option.alternateText, props.task.shareDestination);
+            Navigation.goBack();
+        }
+    };
+
+    return (
+        <ScreenWrapper includeSafeAreaPaddingBottom={false}>
+            {({didScreenTransitionEnd, safeAreaPaddingBottomStyle}) => (
+                <>
+                    <HeaderWithCloseButton
+                        title={props.translate('newTaskPage.assignee')}
+                        onCloseButtonPress={() => Navigation.goBack()}
+                        shouldShowBackButton
+                        onBackButtonPress={() => Navigation.goBack()}
+                    />
+                    <View style={[styles.flex1, styles.w100, styles.pRelative]}>
+                        <OptionsSelector
```

Comment:
Hi all, this PR caused issue https://github.com/Expensify/App/issues/20075

The asignee selection list can take a long time to filter. So we should have debounced search when implementing OptionsSelector [(like done in this PR)](https://github.com/Expensify/App/pull/22095/files)